teps/0110-decouple-catalog-organization-and-reference.md

```diff
@@ -0,0 +1,235 @@
+For example, `https://api.hub.tekton.dev/v1/resource/Tekton/task/buildpacks/0.4` hosts the *buildpacks* `Task`. But the
+structure of the data there requires users to process the data first before applying the resources to their clusters.
+The new HTTP endpoint mentioned above should serve the *yaml* files that users can easily install resources to their
+clusters, similarly the current experience using GitHub HTTP endpoint:
+
+```shell
+# Current: using GitHub path in existing Tekton Catalog organization
+kubectl apply -f https://raw.githubusercontent.com/tektoncd/catalog/main/task/buildpacks/0.4/buildpacks.yaml
+
+# Proposal: using the HTTP Endpoint from Tekton Catalog - should serve same yaml as the one above
+kubectl apply -f https://hub.tekton.dev/tekton//task/buildpacks/0.4
+```
+
+#### Remote Resolution - Hub Resolver
+
+We propose that Tekton hosts and maintains the [Hub Resolver][hub-resolver] that would be used for remote resolution of
+resources from the Tekton Hub. For example, the `TaskRun` below uses the Hub Resolver to fetch the *buildpacks* `Task`:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: buildpacks
+spec:
+  taskRef:
+    resolver: hub
+    resource:
+    - name: kind
+      value: task
+    - name: name
+      value: buildpacks
+    - name: version
+      value: "0.4"
+```
+
+For more information on remote resolution, see [TEP-0060: Remote Resource Resolution][tep-0060].
+
+#### CLI
+
+Use of the Hub CLI is not tightly coupled to the Catalog organization. Therefore, the users of the Hub CLI can continue
+using it to install resources, even after we reorganize the Catalog - the updates needed will be implementation details.
+
+```shell
+tkn hub install task buildpacks --version 0.4 --from tekton
+```
+
+### Tekton Catalog
+
+#### Remote Resolution - Git Resolver
+
+Today, Tekton hosts and maintains the [Git Resolver][git-resolver] that is used for remote resolution of resources
+from the Tekton Catalog. It takes an inputs `url`, `branch` and `path` to resolve the resource the user wants to
+install from the Catalog, as shown below:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: buildpacks
+spec:
+  pipelineRef:
+    resolver: git
+    resource:
+    - name: url
+      value: https://github.com/tekton/catalog.git
+    - name: branch
+      value: main
+    - name: path
+      value: task/buildpacks/0.1/buildpacks.yaml
+```
+
+With the above approach, users could define their `PipelineRuns` and `TaskRuns` to rely on the current directory format.
+We propose removing the `path` input in the Git Resolver, and replacing it with `kind`, `name` and `version`. With this
+approach, users would not rely on the directory structure - the Git Resolver will handle resolving the paths as needed.
+The above example would be updated as such:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: buildpacks
+spec:
+  pipelineRef:
```

Comment:
`PipelineRef` --> `TaskRef`?

Reply:
good catch, yes meant to write `taskRef`

cc @vdemeester